assets/js/components/OverlayNotification/LinkAnalyticsAndAdSenseAccountsOverlayNotification.test.js

```diff
@@ -0,0 +1,298 @@
+/**
+ * LinkAnalyticsAndAdSenseAccountsOverlayNotification component tests.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import LinkAnalyticsAndAdSenseAccountsOverlayNotification, {
+	LINK_ANALYTICS_ADSENSE_OVERLAY_NOTIFICATION,
+} from './LinkAnalyticsAndAdSenseAccountsOverlayNotification';
+import {
+	render,
+	createTestRegistry,
+	provideModules,
+	fireEvent,
+	act,
+} from '../../../../tests/js/test-utils';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
+import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
+import { VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY } from '../../googlesitekit/constants';
+
+describe( 'LinkAnalyticsAndAdSenseAccountsOverlayNotification', () => {
+	let registry;
+
+	const fetchGetDismissedItems = new RegExp(
+		'^/google-site-kit/v1/core/user/data/dismissed-items'
+	);
+	const fetchDismissItem = new RegExp(
+		'^/google-site-kit/v1/core/user/data/dismiss-item'
+	);
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		provideModules( registry, [
+			{
+				slug: 'adsense',
+				active: true,
+				connected: true,
+			},
+			{
+				slug: 'analytics-4',
+				active: true,
+				connected: true,
+			},
+		] );
+		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
+			adSenseLinked: false,
+		} );
+	} );
+
+	it( 'does not render when Analytics module is not connected', () => {
```

Comment:
Should there be another test for AdSense module as well?

Reply:
Oh, true, I'll add that 👍🏻